library/src/interactor_impl.cxx

```diff
@@ -71,8 +75,90 @@ class interactor_impl::internals
 #endif
   }
 
+  //----------------------------------------------------------------------------
+  // Method defined to normalize the Z axis so all models are treated temporarily
+  // as Z-up axis models.
+  void zUpTransforms(vtkMatrix3x3* to, vtkMatrix3x3* from)
+  {
+    vtkRenderer* renderer =
+      this->VTKInteractor->GetRenderWindow()->GetRenderers()->GetFirstRenderer();
+    const double* up = renderer->GetEnvironmentUp();
+    const double* right = renderer->GetEnvironmentRight();
+    double fwd[3];
+    vtkMath::Cross(right, up, fwd);
+    const double m[9] = {
+      right[0], right[1], right[2], //
+      fwd[0], fwd[1], fwd[2],       //
+      up[0], up[1], up[2],          //
+    };
+    to->DeepCopy(m);
+    from->DeepCopy(m);
+    from->Transpose();
+  }
+
+  //----------------------------------------------------------------------------
+  // Set the view orbit position on the viewport.
+  enum class ViewType
+  {
+    VT_FRONT,
+    VT_RIGHT,
+    VT_TOP,
+    VT_ISOMETRIC
+  };
+  static void SetViewOrbit(ViewType view, internals* self)
+  {
+    vtkNew<vtkMatrix3x3> toZup, fromZup;
+    self->zUpTransforms(toZup, fromZup);
+    camera& cam = self->Window.getCamera();
+    vector3_t up = { 0, 0, 1 };
+    point3_t pos = cam.getPosition();
+    point3_t foc = cam.getFocalPoint();
+    point3_t axis, newPos;
+
+    /* convert coords to +Z up */
+    toZup->MultiplyPoint(pos.data(), pos.data());
+    toZup->MultiplyPoint(foc.data(), foc.data());
+
+    const double dx = foc[0] - pos[0];
+    const double dy = foc[1] - pos[1];
+    const double dz = foc[2] - pos[2];
+    double radius = sqrt(dx * dx + dy * dy + dz * dz);
+    switch (view)
+    {
+      case ViewType::VT_FRONT:
+        axis = { 0, 0, +1 };
+        break;
+      case ViewType::VT_RIGHT:
+        axis = { +1, 0, 0 };
+        break;
+      case ViewType::VT_TOP:
+        axis = { 0, +1, 0 };
+        up = { 0, -1, 0 };
+        break;
+      case ViewType::VT_ISOMETRIC:
+        axis = { -1, +1, +1 };
+        break;
+    }
+    fromZup->MultiplyPoint(up.data(), up.data());
+    fromZup->MultiplyPoint(axis.data(), axis.data());
+
+    newPos[0] = foc[0] + radius * axis[0];
+    newPos[1] = foc[1] + radius * axis[1];
+    newPos[2] = foc[2] + radius * axis[2];
+
+    /* convert coordinates back to whatever up is according to model/options */
+    fromZup->MultiplyPoint(newPos.data(), newPos.data());
```

Comment:
shouldn't need to transform `newPos` because you've just built it above by combining `foc` and `axis` that are both already transformed by `fromZup`

Reply:
Fixed after @Meakk's PR: https://github.com/f3d-app/f3d/pull/863/commits/58f6f99b8acc4e003345d13825d2fbe6c85ec649